src/Ng/CommandHelpers.cs

```
@@ -12,381 +13,170 @@
-        {
-            if (!arguments.TryGetValue(Constants.ArgumentPrefix + searchArg, out value))
-            {
-                if (required)
-                {
-                    TraceRequiredArgument(searchArg);
-                    throw new ArgumentException("Required argument not provided", searchArg);
-                }
-
-                value = null;
-                return false;
-            }
-
-            return true;
-        }
-
-        private static SecretInjector GetSecretInjector(IDictionary<string, string> arguments)
+        private static ISecretInjector GetSecretInjector(IDictionary<string, string> arguments)
         {
             ISecretReader secretReader;
-            string vaultName;
-            if (!TryGetArgument(arguments, Constants.VaultName, out vaultName))
+            try
             {
-                secretReader = new EmptySecretReader();
+                var vaultName = arguments.GetOrNull(Arguments.VaultName);
+                var clientId = arguments[Arguments.ClientId];
+                var certificateThumbprint = arguments[Arguments.CertificateThumbprint];
+                var storeName = arguments.GetOrNull<StoreName>(Arguments.StoreName) ?? StoreName.My;
+                var storeLocation = arguments.GetOrNull<StoreLocation>(Arguments.StoreLocation) ?? StoreLocation.LocalMachine;
+                var shouldValidateCert = arguments.GetOrThrow<bool>(Arguments.ValidateCertificate);
+
+                var keyVaultConfig = new KeyVaultConfiguration(vaultName, clientId, certificateThumbprint, storeName, storeLocation, shouldValidateCert);
+
+                secretReader = new CachingSecretReader(new KeyVaultReader(keyVaultConfig),
+                    arguments.GetOrNull<int>(Arguments.RefreshIntervalSec) ??
+                    CachingSecretReader.DefaultRefreshIntervalSec);
             }
-            else
+            catch (Exception)
```

> (Exception) [](start = 17, length = 12)

this hides all exceptions from  the user. If there was an initialization error, I want to know about it. Please leave the previous logic as is.